org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/json/adapters/EitherTypeAdapter.java

```diff
@@ -210,13 +210,19 @@ protected Either<L, R> create(JsonToken nextToken, JsonReader in) throws IOExcep
 		throw new JsonParseException("Unexpected token " + nextToken + ": expected " + left + " | " + right + " tokens.");
 	}
 
+	/**
+	 * @throws IOException
```

Comment:
Not sure about suppressing these unused throws warnings in this way. I think that at least `Either.createLeft/createRight` should not throw `IOException`, but this probably also applies to `EitherTypeArgument.read(JsonElement)` and `StreamMessageProducer.handleMessage`.

Reply:
I thought about that too - but I didn't want to introduce an API breaking change to fix these warnings. If we want to do that, it should be pulled out into a different PR and be documented in API breaking changes.

Reply:
OK - will do.